### PR TITLE
CosmosMsg types for multisig client

### LINKED ISSUE #547: Specify CW3 Multisig message types
We currently allow a very loose type for msgs in `Cw3CosmWasmClient`:
https://github.com/cosmos/cosmjs/blob/246da8f4d79d8ced403660df88b714c6584098d3/packages/cosmwasm/src/cw3cosmwasmclient.ts#L32

We should restrict the types according to https://github.com/CosmWasm/cosmwasm/blob/v0.12.0/packages/std/src/results/cosmos_msg.rs#L13-L23 (and allow anything under `custom`.

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #547

### CODE REVIEW

File: packages/cosmwasm/src/msgs.ts [hunk 1]
```diff
@@ -144,3 +144,75 @@ export interface MsgMigrateContract extends Msg {
...
+    readonly send: readonly Coin[];
+    readonly label?: string;
+  };
+}
+
+export interface WasmMsg {
+  readonly wasm: WasmExecuteMsg | WasmInstantiateMsg;
+}
+
+export type CosmosMsg = BankMsg | CustomMsg | StakingMsg | WasmMsg;
```
Comment (webmaster128) [inline packages/cosmwasm/src/msgs.ts:218]: Can you link the definition of the type in a doc comment (https://github.com/CosmWasm/cosmwasm/blob/v0.12.0/packages/std/src/results/cosmos_msg.rs#L10-L23)?

File: packages/cosmwasm/src/msgs.ts [hunk 2]
```diff
@@ -144,3 +144,75 @@ export interface MsgMigrateContract extends Msg {
 export function isMsgMigrateContract(msg: Msg): msg is MsgMigrateContract {
   return (msg as MsgMigrateContract).type === "wasm/MsgMigrateContract";
 }
+
```
Comment (webmaster128) [inline packages/cosmwasm/src/msgs.ts:147]: All those messages 👇 live on a different level than the existing ones 👆. What we have above is sent to the chain. The new types are sent to the contract. I have no great naming suggestion that is better than the current naming, but let's put them in a different file to indicate those are different things.

File: packages/cosmwasm/src/cosmosmsg.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  };
+}
+
+export interface WasmMsg {
+  readonly wasm: WasmExecuteMsg | WasmInstantiateMsg;
+}
+
+/** These definitions are derived from CosmWasm:
+ * https://github.com/CosmWasm/cosmwasm/blob/v0.12.0/packages/std/src/results/cosmos_msg.rs#L10-L23
+ * */
```
Comment (webmaster128) [inline packages/cosmwasm/src/cosmosmsg.ts:76]:
```suggestion
 */
```

File: packages/cosmwasm/types/cosmosmsg.d.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    readonly send: readonly Coin[];
+    readonly label?: string;
+  };
+}
+export interface WasmMsg {
+  readonly wasm: WasmExecuteMsg | WasmInstantiateMsg;
+}
+/** These definitions are derived from CosmWasm:
+ * https://github.com/CosmWasm/cosmwasm/blob/v0.12.0/packages/std/src/results/cosmos_msg.rs#L10-L23
+ * */
```
Comment (webmaster128) [inline packages/cosmwasm/types/cosmosmsg.d.ts:63]:
```suggestion
 */
```

**Review by webmaster128 — APPROVED** [review]

**Review by ethanfrey — COMMENTED** [review]
Nice one

**Pull request merged**